### ISSUE TITLE
Issue with aria-modal="true" role="dialog", after announcing everything, even though we've not tabbed away, NVDA focus on last announced element

### ISSUE THREAD

**Issue by BojanSlavkovskii** [nvaccess/nvda#17650]
### NVDA logs, crash dumps and other attachments:
Uploading video is failing so hopefully this image is fine. As you can see we have only one function in HTML (that is calling the alert) other two are just closing the dialog. Initial focus is on X (we have cdkInitialFocus added but cannot be seen in this image) that works fine, everything is announced, as you can see I'm not tabbing away from the X, that's all NVDA announcements, and when I press enter, focus is moving to X and calling that function.
![Image](https://github.com/user-attachments/assets/174ecd77-586c-47a0-b5ee-492792ca395a)

#### Does the issue still occur after restarting your computer?
- Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
- Yes, it works the same with every version of NVDA.

#### If NVDA add-ons are disabled, is your problem still occurring?
- Yes

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
- Yes

**Comment by BojanSlavkovskii** [comment]
Video of NVDA ( this link will expire in 2 days ): https://streamable.com/sgjob0.

**Comment by Adriani90** [comment]
This is because NVDA enabled browse mode on the modal dialog when entering it. Browse mode allows for a more efficient navigation via quick navigation keys (h for heading, b for buttons etc.). Thus, in this case NVDA uses its virtual cursor instead of the system focus, and pressing enter simulates a mouse click event. If you want to use only the system focus when entering web pages and dialogs, then you can disable browse mode on page load in NVDA browse mode settings category.

Closing as works for me.

**Comment by Adriani90** [comment]
When NVDA enabled browse mode after entering the dialos, a say all is started, meaning the page is being read from top to botom. You can also disable say all after page load in NVDA browse mode settings category. In this case the NVDA virtual cursor should stay at the same position as the system focus after page load or after entering a dialog.
